File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution(object):
    def minimumDifference(self, nums, k):
        if k <= 1:
            return 0
        nums.sort()
        mins = float('inf')
        for i in range(len(nums) - k + 1):
            mins = min(mins, nums[i+k-1]- nums[i])
        return mins

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        if(k <= 1) return 0;
        int mins = Integer.MAX_VALUE;
        Arrays.sort(nums);
        for(int i = 0; i <= nums.length-k; i++){
            mins = Math.min(mins,nums[i+k-1]- nums[i]);
        }
        return mins;
    }
}